internal/storage/metrics_mem.go

```diff
@@ -0,0 +1,61 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+	"sync"
+)
+
+// MemAccountStore is a memory implementation of the account store
+// to use for testing purposes, etc.
+type MemAccountStore struct {
+	metrics map[string][]UserLoginMetric
+	mu      sync.Mutex
+}
+
+func NewMemAccountStore() *MemAccountStore {
+	return &MemAccountStore{
+		metrics: map[string][]UserLoginMetric{},
+	}
+}
+
+func (mas *MemAccountStore) WroteMetric(accountId, userId string) (bool, error) {
+	mas.mu.Lock()
+	metrics, ok := mas.metrics[accountId]
+	mas.mu.Unlock()
+
+	if !ok {
+		return false, fmt.Errorf("no metrics found for account")
+	}
+
+	for _, metric := range metrics {
+		if metric.UserID == userId {
+			return true, nil
+		}
+	}
+
+	return false, fmt.Errorf("could not find write for account id: %s, user_id: %s", accountId, userId)
+}
+
+func (mas *MemAccountStore) Write(ctx context.Context, metric UserLoginMetric) error {
+	mas.mu.Lock()
+	defer mas.mu.Unlock()
+	if _, ok := mas.metrics[metric.AccountID]; !ok {
+		mas.metrics[metric.AccountID] = []UserLoginMetric{metric}
+		return nil
+	}
+
+	metrics := mas.metrics[metric.AccountID]
```

Comment:
`append` will work when `metrics` is nil
```suggestion
	metrics := mas.metrics[metric.AccountID]
```